### DISCUSSION TITLE
DIM_GET sample code with parent child hierarchy with duplicate elements

### DISCUSSION THREAD

**Discussion by SatyaDnv** [cubewise-code/tm1py#783] [General]
**Describe what did you try to do with TM1py**
sample folder dim_get.py

**Describe what's not working the way you expect**
Didn't get the expected result? Describe: The result provides parent, chld.
1. can we get parentID, Parentelement, ChildID, child element, uniquepath. A sample python code to get all the 5 columns.
2. to build a hierarchy in SQL or viz tools.

**Comment by MariusWirtz** [comment]
Last year we introduced a function to TM1py that retrieves hierarchy information in a shape that is convenient for UI tools like PowerBI.
You may find this function useful.

``` python
from TM1py import TM1Service

with TM1Service(address='localhost', port=12354, ssl=True, user="admin", password="apple") as tm1:
    df = tm1.power_bi.get_member_properties(
        dimension_name="Customer",
        hierarchy_name="Customer",
        member_selection="{TM1SubsetAll([Customer])}",
        attributes=["City", "State", "Country/Region", "Longitude", "Latitude"],
        skip_consolidations=True
    )

    print(df.head(2).to_markdown())

```

|    |   Customer | Type    | City    | State   | Country/Region   |   Longitude |   Latitude | level000       |
|---:|-----------:|:--------|:--------|:--------|:-----------------|------------:|-----------:|:---------------|
|  0 |       1023 | Numeric | Irving  | TX      | US               |    -96.9969 |    32.8748 | Total Customer |
|  1 |      10000 | Numeric | Chicago | IL      | US               |    -87.6221 |    41.8853 | Total Customer |
  

it retuns a pandas dataframe so you can easily write it to SQL with `df.to_sql`
